app/frontend/actions.js

```diff
@@ -233,14 +245,26 @@ module.exports = ({setState, getState}) => {
     const balance = await wallet.getBalance()
     const ownAddressesWithMeta = await wallet.getOwnAddressesWithMeta()
     const transactionHistory = await wallet.getHistory()
+    const conversionRates = getConversionRates(state)
 
     // timeout setting loading state, so that loading shows even if everything was cached
     setTimeout(() => setState({loading: false}), 500)
     setState({
       balance,
       ownAddressesWithMeta,
       transactionHistory,
+      conversionRates,
```

Comment:
this is a promise if I understand right, I'm not sure if you want to have a promise there. I think it would be safer to not set the conversionRates there at all and you could move the "const conversionRates = getConversionRates(state)" inside of the try-catch block below then.

I see one particular upside in the approach I suggest - when fetching fails, it does not reset your conversion rates to null when they were set previously. It probably makes sense to keep an older rate rather than delete it completely.